### ISSUE TITLE
Sent emails stay on draft box

### ISSUE THREAD

**Issue by re-nobre** [pixelated/pixelated-user-agent#512]
## What
Sometimes, when I sent an email a partial draft keep on draft box an I need to delete it by myself.

## Definition of done
When an email will be send 
All its drafts should be deleted

**Comment by shyba** [comment]
We just discovered that this was a design issue.
Creating and updating drafts should rely on Message-Id header as rfc 2822.
This will be modified to use it as a draft unique identifier.
leap_mail already supports it (even has a index for that).

**Comment by pereiragislene** [comment]
We've changed the code to use Message-Id as unique identifier for the mails, but now we are facing a drafts duplication problem while editing the drafts.
While doing some research about this issue, we discovered that the drafts duplication problem also happens in Mail clients as Thunderbird and Apple Mail (e.g.: https://bugzilla.mozilla.org/show_bug.cgi?id=402132, https://www.ddns.com.au/knowledgebase/133/Why-do-I-have-duplicate-items-in-drafts-folder-when-using-IMAP.html, http://www.macobserver.com/tmo/answers/how_to_configure_apple_mail_to_avoid_duplicate_drafts_with_gmail).
We are still investigating workarounds that we can use to fix the problem.
